files/build_templates/docker_image_ctl.j2

```diff
@@ -424,20 +424,23 @@ start() {
         if [[ $? -ne 0 ]] ; then
             echo "sonic-cfggen failed to render asic_table.json"
             echo "$(cat errorlog.txt)"
+            sonic-cfggen -a '{"DEVICE_METADATA":{"localhost":{"platform":"'$PLATFORM'"}}}' -t /usr/share/sonic/templates/asic_table.j2 > /etc/sonic/asic_table.json
```

Comment:
>/usr/share/sonic/templates/asic_table.j2 [](http://example.com/codeflow?start=94&length=40)

If someone change this j2 and using extra var other than "platform", how do you detect the breaking change?

Currently it will fail the command and silently continue. Is it better to report the error and exit with an error code?

Reply:
If a vendor updated the template with another variable introduced, the failure can be detected by analyzing the dump. In that case, the asic_table.json will be an empty list `[]` and the root cause can be easily understood. In this sense, it doesn't bring too much benefit to detect the failure as this is a WA.
In addition, `sonic-cfggen` won't return an error if a variable isn't defined because sometimes it is expected from the j2 template's PoV.